src/tests/test_lua_ptr.cpp

```diff
@@ -33,11 +33,26 @@ BOOST_AUTO_TEST_CASE(test_lua_ptr) {
 	BOOST_CHECK(ptr);
 	BOOST_CHECK_EQUAL(ptr.get_ptr(), &obj);
 	{
-		auto obj_moved = std::move(obj);
+		// test move constructor
+		auto obj2 = std::move(obj);
 		BOOST_CHECK(ptr);
-		BOOST_CHECK_EQUAL(ptr.get_ptr(), &obj_moved);
+		BOOST_CHECK_EQUAL(ptr.get_ptr(), &obj2);
 		BOOST_CHECK_EQUAL(ptr->value, "test");
+
+		lua_ptr<dummy_object> ptr_from_moved_object(obj);
+		BOOST_CHECK(!ptr_from_moved_object);
```

Comment:
The build failure on Ubuntu was clang-tidy warning about a use-after-move here. Will add a `NOLINT` for it.